packages/components/table/src/store/watcher.ts

```diff
@@ -202,6 +201,49 @@ function useWatcher<T>() {
       instance.emit('selection-change', newSelection)
     }
   }
+  const toggleRowStatus = <T>(
+    statusArr: T[],
+    row: T,
+    newVal: boolean
+  ): boolean => {
+    let changed = false
+    const index = statusArr.indexOf(row)
+    const included = index !== -1
+
+    const addRow = () => {
+      statusArr.push(row)
+      if (row.children) {
```

Comment:
IMO, the children may not be an array. Maybe it should be judged. What do you think?

```suggestion
      if (isArray(row.children)) {
```

Reply:
yes, the children type is totally  depend on user